File: MyStoreBack/Models/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace MyStoreBack.Models;

public class RegisterModel
{
    public string Email { get; set; } = String.Empty;
    public string Password { get; set; } = String.Empty;
    [StringLength(100), Required]
    public string FirstName { get; set; } = String.Empty;
    [StringLength(100), Required]
    public string LastName { get; set; } = String.Empty;
    [StringLength(300)]
    public string? Image { get; set; }
}

File: MyStoreClient/app/src/main/java/com/example/mystore/models/user/RegisterModel.kt
package com.example.mystore.models.user

import com.example.mystore.models.Helper.createPartFromString
import okhttp3.MediaType
import okhttp3.RequestBody

data class RegisterModel(
    var email: String = "",
    var userName: String = "",
    var password: String = "",
    var firstName: String = "",
    var lastName: String = "",
) {
    fun toMap(): MutableMap<String, RequestBody> = mutableMapOf(
        "email" to createPartFromString(email),
        "userName" to createPartFromString(userName),
        "password" to createPartFromString(password),
        "firstName" to createPartFromString(firstName),
        "lastName" to createPartFromString(lastName),
    )
}